VerilogCodingStyle.md

```diff
@@ -1797,92 +1797,167 @@ increment.
 
 ### Don't Cares (`X`'s)
 
-***Explicitly specify don't cares when safe to do so. Don't silently
-squash `X` in logic.***
+***The use of `X` literals in RTL code is strongly discouraged. RTL must not
+assert `X` to indicate "don't care" to synthesis in any case.  In order to flag
+and detect invalid conditions, rather than assign and propagate `X` values,
+designs should fully define all signal values and make extensive use of SVAs to
+indicate the invalid conditions. ***
+
+If not strictly controlled, the use of `X` assignments in RTL to flag invalid or
+don't care conditions can lead to simulation/synthesis mismatches.
+
+Instead of assigning and propagating `X` in order to flag and detect invalid
+conditions, it is encouraged to make **extensive use of SVAs**. The added
+benefits of this design practice are that:
+
+- No special code style is required to properly propagate `X` conditions,
+- The chance of accidentally introducing simulation/synthesis mismatches is
+  systematically reduced,
+- Simulation fails quickly and less signal backtracking is needed to root-cause
+  bugs,
+- In several cases, formal property verification (FPV) can be used to prove
+  whether these SVAs can always be fulfilled,
+- In a security context, deterministic/defined behavior is desired, even for
+  illegal/invalid/unreachable input combinations (sometimes stated more tersely
+  as "for security-critical designs, there are no don't-cares").
+
+The solution presented here has similarities with the approaches presented in
+["Being Assertive With Your X"](http://www.lcdm-eng.com/papers/snug04_assertiveX.pdf)
+by Don Mills.
+
+Note that although don't cares can be used to indicate possible optimization
+opportunities to the synthesis tool, it is debatable whether the gains in logic
+reduction are significant enough to outweigh the possible synthesis mismatch
+issues that the use of `X` literals may entail (especially with the gate-counts
+available in today's technologies). Large LUTs may be an exception to this
+statement and the use of `X` in that context should be carefully evaluated on a
+case-by-case basis.
+
+#### Specific Guidance on Case Statements and Ternaries
+
+To comply with this style, RTL must place `` `ASSERT_KNOWN`` assertions on all
+module outputs, with the exception of signals that may implicitly be `X` at the
+beginning of the simulation, such as FIFO, SRAM or register file outputs.
```

Comment:
In the separate security guidelines we way want to say all such structures should receive an explicit reset to 0 via FSM during an init phase?

Reply:
That is another question for @cdgori :)...